components/admin_console/team_channel_settings/group/group_users/group_users.tsx

```diff
@@ -0,0 +1,393 @@
+                            id='admin.user_item.member'
+                            defaultMessage='Member'
+                        />
+                    ),
+                    value: false,
+                },
+                [GeneralConstants.CHANNEL_ADMIN_ROLE]: {
+                    name: (
+                        <FormattedMessage
+                            id='admin.user_grid.channel_admin'
+                            defaultMessage='Channel Admin'
+                        />
+                    ),
+                    value: false,
+                },
+            };
+            filterOptions.role.keys = [GeneralConstants.SYSTEM_GUEST_ROLE, GeneralConstants.CHANNEL_USER_ROLE, GeneralConstants.CHANNEL_ADMIN_ROLE, GeneralConstants.SYSTEM_ADMIN_ROLE];
+        } else if (this.props.scope === 'team') {
+            filterOptions.role.values = {
+                ...filterOptions.role.values,
+                [GeneralConstants.TEAM_USER_ROLE]: {
+                    name: (
+                        <FormattedMessage
+                            id='admin.user_item.member'
+                            defaultMessage='Member'
+                        />
+                    ),
+                    value: false,
+                },
+                [GeneralConstants.TEAM_ADMIN_ROLE]: {
+                    name: (
+                        <FormattedMessage
+                            id='admin.user_grid.team_admin'
+                            defaultMessage='Team Admin'
+                        />
+                    ),
+                    value: false,
+                },
+            };
+            filterOptions.role.keys = [GeneralConstants.SYSTEM_GUEST_ROLE, GeneralConstants.TEAM_USER_ROLE, GeneralConstants.TEAM_ADMIN_ROLE, GeneralConstants.SYSTEM_ADMIN_ROLE];
+        }
+
+        if (!this.props.enableGuestAccounts) {
+            delete filterOptions.role.values[GeneralConstants.SYSTEM_GUEST_ROLE];
+            filterOptions.role.keys.splice(0, 1);
+        }
+
+        const filterKeys = ['role'];
+        const filterProps = {
+            options: {...filterOptions},
+            keys: filterKeys,
+            onFilter: this.onFilter,
+        };
+
+        const placeholderEmpty: JSX.Element = (
+            <FormattedMessage
+                id='admin.member_list_group.notFound'
+                defaultMessage='No users found'
+            />
+        );
+
+        return (
+            <div className='GroupUsers'>
+                <DataGrid
+                    columns={columns}
+                    rows={rows}
+                    loading={this.state.loading}
+                    page={this.state.page}
+                    nextPage={this.nextPage}
+                    previousPage={this.previousPage}
+                    startCount={startCount}
+                    endCount={endCount}
+                    total={total}
+                    search={this.search}
+                    filterProps={filterProps}
```

Comment:
We're constructing a new instance of `filterProps` each time this component re-renders which means that the `DataGrid` will also re-render each time, assuming it's also a PureComponent. Is that okay, or could that cause performance issues?

Reply:
I think DataGrid being re-rendered anytime this is component is re-rendered is kind of inevitable, what would be a better way to pass these props? Just making the object inline? like `filterProps={{options, keys, onFilter}}`?

Reply:
You could consider saving `filterProps` in `this.state` and only modifying it when the props change. `DataGrid` re-rendering may not be the end of the world unless there's something else on this screen that's particularly performance-sensitive, but it's something to keep in mind with more complex components like it.